wallet instance lifecycle (#242)
italia/eidas-it-wallet-docs: docs/en/wallet-attestation.rst

      -
    * - **presentation_definition_uri_supported**
      - Boolean value specifying whether the Wallet Instance supports the transfer of presentation_definition by reference. MUST be set to false.

Wallet Instance Lifecycle

> ------------

Reviewer: ```suggestion
-----------------------------
```

otherwise the linting ...

The ability of the Wallet Instance to obtain a Wallet Attestation is bound to its current state. 

> The Wallet Instance calculates its current state based on its local credentials storage and its revocation state on the Wallet Provider's backend, if present.

Reviewer: ```suggestion
The Wallet Instance assesses its current state based on the Credentials stored locally and the Wallet Attestation issued by the Wallet Provider.
```

> The Wallet Instance lifecycle defines all possible states a Wallet Instance can be in, as well as the transitions between them. The Wallet Instance lifecycle is illustrated in the following diagram:

Reviewer: ```suggestion
The lifecycle of a Wallet Instance encompasses all the potential states it can configure, along with the transitions from one state to another. This lifecycle is depicted in the diagram below:
```

.. figure:: ../../images/wallet_instance_lifecycle.svg
   :name: Wallet Instance Lifecycle

>    :alt: The image illustrates the Wallet Instance lifecycle, with the states explained below.

Reviewer: ```suggestion
   :alt: Illustration representing the Wallet Instance lifecycle, with the states explained below.
```

   :target: https://www.plantuml.com/plantuml/uml/SoWkIImgAStDuOhMYbNGrRLJyCm32kNafAPOAMH2c5mAG00N1YloBqWjIYp9pCzBpB5IA4ijoaoh1Ab25WUh2qlCoKm1gW1HYIMf83KGCKnJClDmg799JKmkoIm3IW1DAaejoyzEHRSBfpfCbmEzQQLGceVaDOH6x4emxS9KWd0mfgH3QbuAC801

> A Wallet Instance can obtain a Wallet Attestation is it's in either `Installed`, `Operational` or `Valid` state; that implies that a `Deactivated` Wallet Instance cannot obtain a Wallet Attestation hence it cannot interact with other actors such as Credentil Issuers and Relying Parties.

Reviewer 2: ```suggestion
A Wallet Instance can obtain a Wallet Attestation if it's in either `Installed`, `Operational` or `Valid` state; that implies that a `Deactivated` Wallet Instance cannot obtain a Wallet Attestation hence it cannot interact with other entities such as PID/(Q)EAA Providers and Relying Parties.
```

Author: thanks

> A Wallet Instance can obtain a Wallet Attestation if it's in either `Installed`, `Operational` or `Valid` state; that implies that a `Deactivated` Wallet Instance cannot obtain a Wallet Attestation hence it cannot interact with other entities such as PID/(Q)EAA Providers and Relying Parties.

Reviewer: ```suggestion
A Wallet Instance SHOULD obtain a Wallet Attestation if it's in either `Installed`, `Operational` or `Valid` state; that implies that a `Deactivated` Wallet Instance cannot obtain a Wallet Attestation hence it cannot interact with other entities of the ecosystem, such as PID/(Q)EAA Providers and Relying Parties.
```

Reviewer 2: This is still not clear. the point is that the Installed, Operational and Valid states depend on a valid Wallet Attestation. This means that if, during the issuance of a Wallet Attestation, something goes wrong and the Wallet Attestation cannot be provided, the state of the Wallet must change. At the same time, a Wallet Instance in a Deactivated state MUST not obtain a valid Wallet Attestation (e.g. this could be the case of a Wallet Instance revocation triggered by the User). Am I missing something?

| State         | Description                                                                                                                    |
+===============+================================================================================================================================+
| `Installed`   | The Holder has installed the Wallet Solution on the device                                                                     |
+---------------+--------------------------------------------------------------------------------------------------------------------------------+
| `Operational` | The Wallet Instance has been verified and the Wallet Hardware Key has been registered; no valid PID is present in the storage. |
+---------------+--------------------------------------------------------------------------------------------------------------------------------+
| `Valid`       | A valid PID is present in the storage                                                                                          |
+---------------+--------------------------------------------------------------------------------------------------------------------------------+

> | `Deactivated` | The Wallet Instance has been revoked and its Wallet Hardware Key has been marked as not usabel                                 |

Reviewer 2: ```suggestion
| `Deactivated` | The Wallet Instance has been revoked and its Wallet Hardware Key has been marked as not usable                                 |
```

Author: thanks

> +---------------+--------------------------------------------------------------------------------------------------------------------------------+

Reviewer: we decided to not represent tables in this way

```suggestion
.. list-table::
   :widths: 20 60
   :header-rows: 1

   * - **State**
     - **Description**
   * - `Installed`
     - The User has installed the Wallet Solution on the device.
   * - `Operational`
     - The Wallet Instance has been verified and the Wallet Hardware Key has been registered; no valid PID is present in the storage.
   * - `Valid`
     - A valid PID is present in the storage.
   * - `Deactivated`
     - The Wallet Instance has been revoked and its Wallet Hardware Key has been marked as not usable.
```

Reviewer: @balanza please take this editorial alignment ^

| `verify`     | The Wallet Instance has been verified by the Wallet Provider and its Wallet Hardware Key has been registered |
+--------------+--------------------------------------------------------------------------------------------------------------+
| `validate`   | The Wallet Instance obtains a valid PID                                                                      |
+--------------+--------------------------------------------------------------------------------------------------------------+
| `invalidate` | The PID expires or gets revoked                                                                              |
+--------------+--------------------------------------------------------------------------------------------------------------+
| `revoke`     | The Wallet Provider marks the Wallet Instance as not usable                                                  |
+--------------+--------------------------------------------------------------------------------------------------------------+
| `uninstall`  | The Holder removes the Wallet Instance from the device                                                       |

> +--------------+--------------------------------------------------------------------------------------------------------------+

Reviewer: ```suggestion
.. list-table::
   :widths: 20 60
   :header-rows: 1

   * - **Transition**
     - **Description**
   * - `install`
     - The User performs a fresh installation or restores the initial state of the Wallet Instance on the device.
   * - `verify`
     - The Wallet Instance has been verified by the Wallet Provider and its Wallet Hardware Key has been registered.
   * - `validate`
     - The Wallet Instance obtains a valid PID.
   * - `invalidate`
     - The PID expires or gets revoked.
   * - `revoke`
     - The Wallet Provider marks the Wallet Instance as not usable.
   * - `uninstall`
     - The User removes the Wallet Instance from the device.
```

Revocations
~~~~~~~~~~~~~~~~~~
As mentioned in the *Wallet Instance initialization and registration* section above, a Wallet Instance is bound to a Wallet Hardware Key and it's uniquely identified by it. 

> The Wallet Provider, which holds the Wallet Hardware Keys, can mark them as *revoked* so to make the related Wallet Instance unusable.

Reviewer: if the Wallet Provider the holder of the wallet hardware keys?

Reviewer: instead of "can mark them as *revoked* so to make the related Wallet Instance unusable." I would say that when the Wallet Instance is not usable anymore (such as ... some example reasons) the Wallet Provider MUST ...

Author: Done in https://github.com/italia/eudi-wallet-it-docs/pull/242/commits/4dbf7e2c2944a30223e7909fc25ff3805a6dd7aa

The details of the revocation mechanism used by the Wallet Provider as well as the data model for maintaining the Wallet Instance references is delegated to the Wallet Provider's implementation.

> During the *Wallet Instance initialization and registration* phase the Wallet Provider may associate the Wallet Instance with a specific Holder uniquely identified within the Wallet Provider's systems as well as with metadata regarding the device the Wallet Instance is running on such as operative system version, chipset capabilities and the Wallet Solution version. 

Reviewer: ```suggestion
During the *Wallet Instance initialization and registration* phase the Wallet Provider MAY associate the Wallet Instance with a specific User. The User SHOULD be uniquely identified as well as with metadata regarding the device the Wallet Instance is running on, the metadata MAY include data related to the operative system and general technical capabilities of the device. 
```

> These informations can allow the Wallet Provider to selectively revoke Wallet Instances based on specific criteria.

Reviewer: ```suggestion
These information allow the User to request the Wallet revocation directly interacting with the Wallet Provider as well as enabling the Wallet Provider to revoke a specific Wallet Instance.
```

> The choice of which data need to be stored is left to the Wallet Provider.

Reviewer: Please provide a non-normative example of these data

Author: There's no example data to be proposed here as it's delegated to the Wallet Provider's implementation.

That is, each Wallet Provider decides what metadata is collected from the Wallet Instance to provide the safest experience to users along with the possibility to intervene in case of known issues.

+---------------+--------------------------------------------------------------------------------------------------------------------------------+
| `Deactivated` | The Wallet Instance has been revoked and its Wallet Hardware Key has been marked as not usable                                 |
+---------------+--------------------------------------------------------------------------------------------------------------------------------+

Transitions
~~~~~~~~~~~~~~~~~~
.. list-table::
   :widths: 20 60
   :header-rows: 1

   * - **Transition**
     - **Description**
   * - `install`
     - The User performs a fresh installation or restores the initial state of the Wallet Instance on the device.
   * - `verify`
     - The Wallet Instance has been verified by the Wallet Provider and its Wallet Hardware Key has been registered.
   * - `validate`
     - The Wallet Instance obtains a valid PID.
   * - `invalidate`
     - The PID expires or gets revoked.
   * - `revoke`
     - The Wallet Provider marks the Wallet Instance as not usable.
   * - `uninstall`
     - The User removes the Wallet Instance from the device.

Revocations
~~~~~~~~~~~~~~~~~~

> As mentioned in the *Wallet Instance initialization and registration* section above, a Wallet Instance is bound to a Wallet Hardware Key and it's uniquely identified by it. 

Reviewer 2: It is better to clarify that only the Wallet Provider is able to identify the Wallet Instance by the Wallet Hardware Key uniquely. It is important to avoid confusion on this aspect.

Author: Added in https://github.com/italia/eudi-wallet-it-docs/pull/242/commits/4dbf7e2c2944a30223e7909fc25ff3805a6dd7aa

> The Wallet Instance shares its Wallet Hardware Key only with the Wallet Provider, thus the Wallet Provider is the only entity that can identify a Wallet Instance by its  Wallet Hardware Key.

| `Deactivated` | The Wallet Instance has been revoked and its Wallet Hardware Key has been marked as not usable                                 |
+---------------+--------------------------------------------------------------------------------------------------------------------------------+

Transitions
~~~~~~~~~~~~~~~~~~
.. list-table::
   :widths: 20 60
   :header-rows: 1

   * - **Transition**
     - **Description**
   * - `install`
     - The User performs a fresh installation or restores the initial state of the Wallet Instance on the device.
   * - `verify`
     - The Wallet Instance has been verified by the Wallet Provider and its Wallet Hardware Key has been registered.
   * - `validate`
     - The Wallet Instance obtains a valid PID.
   * - `invalidate`
     - The PID expires or gets revoked.
   * - `revoke`
     - The Wallet Provider marks the Wallet Instance as not usable.
   * - `uninstall`
     - The User removes the Wallet Instance from the device.

Revocations
~~~~~~~~~~~~~~~~~~
As mentioned in the *Wallet Instance initialization and registration* section above, a Wallet Instance is bound to a Wallet Hardware Key and it's uniquely identified by it. 

> The Wallet Instance shares its Wallet Hardware Key only with the Wallet Provider, thus the Wallet Provider is the only entity that can identify a Wallet Instance by its  Wallet Hardware Key.

Reviewer: ```suggestion
The Wallet Instance SHOULD send its public Wallet Hardware Key with the Wallet Provider, thus the Wallet Provider MUST identify a Wallet Instance by its Wallet Hardware Key.
```

When a Wallet Instance is not usable anymore, the Wallet Provider MUST revoke it. The revocation process is a unilateral action taken by the Wallet Provider, and it MUST be performed when the Wallet Instance is in the `Operational` or `Valid` state.
A Wallet Instance becomes unusable for several reasons, such as: the User requests the revocation, the Wallet Provider detects a security issue, or the Wallet Instance is no longer compliant with the Wallet Provider's security requirements.

The details of the revocation mechanism used by the Wallet Provider as well as the data model for maintaining the Wallet Instance references is delegated to the Wallet Provider's implementation.

> During the *Wallet Instance initialization and registration* phase the Wallet Provider MAY associate the Wallet Instance with a specific User. The User SHOULD be uniquely identified as well as with metadata regarding the device the Wallet Instance is running on, the metadata MAY include data related to the operative system and general technical capabilities of the device. 

Reviewer: ```suggestion
During the *Wallet Instance initialization and registration* phase the Wallet Provider MAY associate the Wallet Instance with a specific User, subject to obtaining the User's consent. The Wallet Provider MUST evaluate the operating system and general technical capabilities of the device to check compliance with the technical and security requirements and to produce the Wallet Instance metadata.
```

> These information allow the User to request the Wallet revocation directly interacting with the Wallet Provider as well as enabling the Wallet Provider to revoke a specific Wallet Instance.

Reviewer: ```suggestion
When the User consents to being linked with the Wallet Instance, they gain the ability to directly request Wallet revocation from the Wallet Provider, and it also allows the Wallet Provider to revoke the Wallet Instance associated with that User.
```

> The choice of which data need to be stored is left to the Wallet Provider.

Reviewer: ```suggestion
```

I would not create conflicts with eIDAS regulation about the data required.

> 
> 

Reviewer: ```suggestion
```
